strax/context.py

```diff
@@ -808,19 +818,64 @@ def get_iter(self, run_id: str,
+                end_time = float('inf')
+                for t in strax.to_str_tuple(targets):
+                    try:
+                        # Sometimes some metadata might be missing e.g. during tests.
+                        chunks = self.get_meta(run_id, t)['chunks']  
+                        start_time = max(start_time, chunks[0]['start'])
+                        end_time = min(end_time, chunks[-1]['end'])
+                    except (strax.DataNotAvailable, KeyError):
+                        progress_bar = False
+
+            # Define nice progressbar format:
+            bar_format = "{desc}: |{bar}| {percentage:.2f} % [{elapsed}<{remaining}],"\
+                         " {postfix[0]} {postfix[1][spc]:.2f} s/chunk,"\
+                         " #chunks processed: {postfix[1][n]}"
+            sec_per_chunk = np.nan  # Have not computed any chunk yet.
+            post_fix = ['Rate last Chunk:', {'spc': sec_per_chunk, 'n': 0}]
+
         try:
-            for result in strax.continuity_check(generator):
-                seen_a_chunk = True
-                if not isinstance(result, strax.Chunk):
-                    raise ValueError(f"Got type {type(result)} rather than "
-                                     f"a strax Chunk from the processor!")
-                result.data = self.apply_selection(
-                    result.data, 
-                selection_str=selection_str,
-                keep_columns=keep_columns,
-                time_range=time_range,
-                time_selection=time_selection)
-                yield result
+            with contextlib.ExitStack() as stack:
+                if progress_bar: 
+                    # Get initial time
+                    pbar = stack.enter_context(tqdm(total=1, postfix=post_fix, bar_format=bar_format))
+                    last_time = pbar.last_print_t       
+                
+                for n_chunks, result in enumerate(strax.continuity_check(generator), 1):
+                    seen_a_chunk = True
+                    if not isinstance(result, strax.Chunk):
+                        raise ValueError(f"Got type {type(result)} rather than "
+                                         f"a strax Chunk from the processor!")
+                    result.data = self.apply_selection(
+                        result.data,
+                        selection_str=selection_str,
+                        keep_columns=keep_columns,
+                        time_range=time_range,
+                        time_selection=time_selection)
+                    
+                    if progress_bar:
+                        # Update progressbar:
+                        pbar.n = (result.end - start_time) / (end_time - start_time)
+                        pbar.update(0)
+                        # Now get last time printed and refresh seconds_per_chunk:
+                        # This is a small work around since we do not know the
+                        # pacmaker here.
```

Comment:
[minor] if you were to do another commit you can correct this to _pacemaker_ (sorry for making these minor comments but I think very few people know what the pacemaker is so it might not ring a bell).

Reply:
Totally fine.